extensions/common/istio_dimensions.h

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <set>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_replace.h"
+#include "absl/strings/string_view.h"
+
+namespace Wasm {
+namespace Common {
+namespace {
+// This map maintains the name of Istio dimension and their key to
+// IstioDimensions::dimensions map.
+absl::flat_hash_map<const absl::string_view, unsigned int> dimensions_name_map;
+}  // namespace
+
+// A structure that can hold multiple Istio dimensions(metadata variables).
+// This could be use to key caches based on Istio dimensions for various
+// filters.
+// TODO: Add support for evaluating dynamic Istio dimensions.
+struct IstioDimensions {
```

Comment:
Can this be merged with the one in stats?

Reply:
This is derived from the stats one... to make it more generics...